src/auto-upgrade/updatePackage.js

```diff
@@ -12,6 +12,23 @@ const PROJECT_PACKAGE_LOCATION = path.join(process.cwd(), "package.json");
 // don't have to do file I/O more than once
 let _projectPackageData;
 
+
+/**
+ * This function is used to put the dependencies in alphabetical order.
+ * It takes a hash and returns the same hash ordered by its keys.
+ *
+ * @param {Object} Unordered hash which one would like to be sorted by keys
+ *
+ * @return {Object}
+ */
+function orderHashByKeys (unordered) {
+  const ordered = {};
```

Comment:
Unfortunately this will not work… The spec for objects in Ecmascript state:
> ### 4.3.3 Object
> An object is a member of the type Object. It is an **unordered collection of properties** each of which contains a primitive value, object, or function. A function stored in a property of an object is called a method.

[Old ES3 Spec where javascript objects are defined](http://www.ecma-international.org/publications/files/ECMA-ST-ARCH/ECMA-262,%203rd%20edition,%20December%201999.pdf)

In my experience, Chrome actually does jumble the order. I'm sure it's for optimization reasons.

That being said, we need to use something else. ES2015 gives us the Map datatype. This is what you want to use: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

From there though, you will need to see what happens when you JSON.stringify a map, I'm actually not sure. It might just work, but it might require some more work on your part.